packages/ses/src/commons.js

```diff
@@ -378,3 +378,36 @@ export const FERAL_STACK_GETTER = feralStackGetter;
  * @type {((newValue: any) => void) | undefined}
  */
 export const FERAL_STACK_SETTER = feralStackSetter;
+
+const getAsyncGeneratorFunctionInstance = () => {
```

Comment:
For the benefit of subsequent reviewers, this is a strategy I favored over duplicating this check. This should be reviewed with an eye for all supported environments including the possibility that CSP bars eval. I would like to page in @Jack-Works to assess whether this damages any invariants they might be relying upon. I also don’t have enough experience with CSP to to trace the behavior implied below. Does this throw or return `undefined`? Is the behavior acceptable?

Reply:
@naugtur reminder to take a look

Reply:
When calling `new Function()` under CSP that prevents it, the error thrown is `EvalError` as per https://www.w3.org/TR/CSP2/
![Screenshot 2024-11-27 at 11-19-30 Content Security Policy Level 2](https://github.com/user-attachments/assets/63c0633e-b12f-4402-8867-b36d4f533b38)
The error will therefore be rethrown. 

The way SES supports evaluators being blocked by CSP is not reaching the code that uses them for the purpose of taming `Function` and its friends, which is ok to skip since under CSP they are unusable anyway (as is the evaluation in a Compartment and the `with` statements have to be inlined by the build tool - in case of browserify and webpack that's exactly what lavamoat does)

The open problem here is: this error will be thrown here because the evaluation of this part is not conditional on CSP detection. 
This change should, if possible, be moved over to the section in code where it's used for the purpose of taming. It would also benefit the explanation of the conditional nature of the `AsyncGeneratorFunctionInstance`

Reply:
TODO: option 3 preferred (discussed in endo sync)

Reply:
> Fall back to obtaining `AsyncGeneratorFunctionInstance` with syntax inside the catch block and eliminate that syntax with our hermes-transform // my opinionated preference

this one ^ pushed the changes, if it's worth adding a `console.warn` or more docs i can do that too